e2e_tests/tests/experiment/experiment.py

```diff
@@ -72,6 +72,84 @@ def cancel_trial(trial_id: int) -> None:
     wait_for_trial_state(trial_id, determinedexperimentv1State.STATE_CANCELED)
 
 
+def wait_for_experiment_by_name_is_active(
+    experiment_name: str,
+    min_trials: int = 1,
+    max_wait_secs: int = conf.DEFAULT_MAX_WAIT_SECS,
+    log_every: int = 60,
+) -> int:
+    for seconds_waited in range(max_wait_secs):
+        try:
+            response = bindings.get_GetExperiments(
+                determined_test_session(), name=experiment_name
+            ).experiments
+            if len(response) == 0:
+                time.sleep(1)
+                continue
+            if len(response) > 1:
+                pytest.fail(
+                    f"Multiple experiments with name={experiment_name}, "
+                    f"expected only 1 experiment."
+                )
+            experiment = response[0]
+            experiment_id = experiment.id
+        # Ignore network errors while polling for experiment state to avoid a
+        # single network flake to cause a test suite failure. If the master is
+        # unreachable multiple times, this test will fail after max_wait_secs.
+        except api.errors.MasterNotFoundException:
```

Comment:
why is this an acceptable error in e2e tests?

Reply:
This is not new. This logic was introduced by Yoav [here](https://github.com/determined-ai/determined/pull/571). If you think this exception handling should be ripped out, we should do it across the board (and look if the number of flakes/spurious test failures increases).

Reply:
Anything based on the `Session` (which includes all of `bindings.*`) already has retries built in, after Ilia added automatic retries [here](https://github.com/determined-ai/determined/pull/4031).

I don't mind ripping out automatic retries in tests across the board.  Yoav's change was so long ago I can't remember what the context was.  Today I would probably say "we should diagnose why network failures are so common we're noticing them in our tests".

But in any case, we don't need duplicate layers of retry logic in tests using the `bindings` module.